polkadot/xcm/xcm-simulator/src/lib.rs

```diff
@@ -445,10 +448,90 @@ macro_rules! decl_test_network {
 			fn deliver(
 				pair: ($crate::Location, $crate::Xcm<()>),
 			) -> Result<$crate::XcmHash, $crate::SendError> {
-				let hash = $crate::fake_message_hash(&pair.1);
+				let hash = $crate::helpers::derive_topic_id(&pair.1);
 				$crate::RELAY_MESSAGE_BUS.with(|b| b.borrow_mut().push_back(pair));
+				$crate::helpers::TopicIdTracker::insert(stringify!($relay_chain), hash.into());
```

Comment:
If we can avoid using the `TopicIdTracker` for getting the expected message hash in all of the tests, maybe we can avoid doing this.

And also maybe we can avoid using  global variables in `TopicIdTracker`. But we can discuss about this in a future review iteration. For now let's see if we can avoid using `TopicIdTracker` in as many tests as possible.